### PR TITLE
Inter thread communication change

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Contributes to VST3 statelessness

The receipt of the "echo" of settings changes by the main thread is generalized with a virtual function of EffectInstance,
which may be overridden to avoid copying everything.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-components/EffectInterface.h [hunk 1]
```diff
@@ -481,6 +474,13 @@ class COMPONENTS_API EffectInstance
    //! Function that has not yet found a use
    //! Correct definitions of it will likely depend on settings and state
    virtual size_t GetTailSize() const;
+
+   //! Main thread receives updates to settings from a processing thread
+   /*!
+    Default implementation simply assigns by copy, not move
+    This might be overridden to copy contents only selectively
+    */
```
Comment (crsib) [inline libraries/lib-components/EffectInterface.h:482]: Why do you require R-vaue here in case you expect copy assignment?

Comment (Paul-Licameli) [reply libraries/lib-components/EffectInterface.h:482]: Allowing more freedom to overriding methods if it is needed

Comment (crsib) [reply libraries/lib-components/EffectInterface.h:482]: You force very harsh contract on the users of this code: they have to provide R-value to this function.

**Comment by Paul-Licameli** [comment]
These changes will be incorporated in #3396

**Pull request closed without merging**